### ISSUE TITLE
grunt unittest task does not work

### ISSUE THREAD

**Issue by romanzenka** [igvteam/igv.js#699]
I ran into multiple issues:
- grunt uses PhantomJS to run qunit tests. Sadly, PhantomJS does not support newer browser features, like support for 'const' and 'let' keywords. That causes files not to parse, which quietly causes chunks of igv object missing
- there is no Promise support within PhantomJS
- PhantomJS seems to no longer be actively developed - people are switching to Puppeteer

So it looks like the entire unit test setup is basically impossible to run headlessly, which leads to tests breaking without anyone noticing.

The least invasive way of fixing this is likely to switch to puppeteer (?) otherwise you would probably have to change setup of your unit tests

**Comment by jrobinso** [comment]
OK, thanks for the investigation.     I'm going to close this as I have no time to pursue this, and there is no expectation currently that the tests can be run headlessly.

**Comment by eweitz** [comment]
I have encountered this as well.  Not being able to run igv.js's tests in a headless web browser essentially blocks progress towards continuous integration (CI), part of #674.

I began transitioning tests to headless Chrome in https://github.com/eweitz/igv.js.  That's how I've run tests for [Ideogram.js](https://github.com/eweitz/ideogram), ever since PhantomJS was abandoned and stopped working with newer versions of macOS a few years ago.  Most of the JavaScript ecosystem uses a similar approach.  Modernizing the tests is not a trivial task, but it would benefit igv.js in the long term by enabling CI and easier external contributions.  

My current plan in #674 is to implement support for ES6 export with the current traditional test suite -- and back out my partial transition, at least temporarily.  (Coincidentally, I will be starting that this week.) Then, after support for ES6 import lands, I would enable headless tests to address that prerequisite for CI.

In other words, if the IGV team is open to transitioning tests to headless Chrome, I would be interested in fixing this issue.

**Comment by jrobinso** [comment]
@eweitz We seem to be discussing this on separate issues.   The unit tests are not a priority for me, you can consider this project to not have unit tests.   They are not there for regression or as part of continuous integration.   What's there mainly tests file parsing,  they in no way test even a small fraction of igv.js functions.   So investing time in that is not worthwhile.

WRT ES6,  I investigated for an hour or so and it looks like it would be easy to make igv.js an ES6 module with a few statements added to "wrapper/header.js",  in the same way node and AMD modules are supported.

**Comment by eweitz** [comment]
Hi Jim, I'll follow up over in #674.
